integrationtests/src/main/java/io/kroxylicious/proxy/testkafkacluster/KafkaClusterConfig.java

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.kroxylicious.proxy.testkafkacluster;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.TreeMap;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+import org.apache.kafka.clients.CommonClientConfigs;
+import org.apache.kafka.common.Uuid;
+import org.apache.kafka.common.config.SaslConfigs;
+import org.junit.jupiter.api.TestInfo;
+
+import io.kroxylicious.proxy.testkafkacluster.KafkaClusterConfig.KafkaEndpoints.Endpoint;
+import lombok.Builder;
```

Comment:
I don't know lombok's Builder support. Strimzi uses sundrio's builder support, which supports nested objects. Whether we used the sundrio builder or not, I think support for nested objects in the builder could be useful. It could, for example, allow declarative creation of topics by the framework. That would alleviate test authors from having to do it with the admin client themselves. Likewise sending some records to a topic. In the extreme case, do we even need to declare a cluster every time, or could the framework just spin up a default one when (just) declarative topics were being used.

To be clear: I'm not suggesting any of this for this PR. I'm just wondering how easy we could make writing tests in general using Kafka.

Reply:
+1 to the idea, but let's leave for a future PR.